README.md

```diff
@@ -1,54 +1,73 @@
+
+An initial exploratory data analysis performed as a part of the release v0.0.1 can be found [here](https://github.com/UBC-MDS/contraceptive_method_predictor/blob/main/src/eda.ipynb)
 
-Given there are a mix of continous and categorical variables, and the outcome is a multiclass problem. We will start our approach by choosing a non linear model such as DecisionTree as our baseline as it handles mutliclass problems efficiently. Saying which for the intital stage of the project we cannot fix on a particular model until we try out different algorithms, other feature engineering steps required and hyperparameter optimization. The levels of our target i.e `choice of contraceptive method` have been categorized into : 1=No-use, 2=Long-term, 3=Short-term. We will use F1 score as our evaluation metric as we want to give equal importance to all levels in the target. 
+## Data Analysis Workflow
 
-After selcting our final model, we will refit the model on the complete training data set, and evaluate its performance on unseen test data set. An overall performance of the model with misclassificaton errors will be presented in the form of a classification report. 
+In this project, we created 6 scripts to run our data analysis. As illustrated in the figure below.
 
-An intital exploratory data analysis performed as a part of the release v0.0.1 can be found [here](https://github.com/UBC-MDS/contraceptive_method_predictor/blob/main/src/eda.ipynb)
+![Figure 1. Data Analysis Flow Chart](doc/flow_chart.jpg)
+
+1.  **src/download_data.py** - A first script that downloads the data from the UCI repository and saves it as a csv
+2.  **src/split_data.py** - A second script that takes in the data generated from first script and save the partitioned data (train data and test data) in csv format.
+3.  **src/eda.py** - A third script that reads the train data, creates plots for the provided training data file for analysis.
+4.  **src/preprocess_model_selection.py** - A fourth script that reads the train data and performs preprocessing, builds a model and render the model file and cross validation result.
+5.  **src/predict.py** - A fifth script that takes in the model file generated from the fourth script as well as test data generated from the second script and generate the test results.
+6.  **doc/countraceptive_method_predictor_report.Rmd** - A sixth script that presents the key exploratory data analysis from the third script as well as the test result summaries from the fifth script.
 
 ## Usage
 
-To replicate the analysis, clone this GitHub repository, install the depndencies listede below, and run the following command at the commandline / terminal from the root directory of this project:
+To replicate the analysis, clone this GitHub repository, install the dependencies listed below, and run the following command at the command line / terminal from the root directory of this project:
```

Comment:
We can say - create the conda environment using the yaml file [here] (url to yaml file in our repository)
`conda env create -f env-cmp.yaml` 
Instead of - "install the dependencies listed below"